nixos/modules/tasks/network-interfaces.nix

```diff
@@ -1055,11 +1055,13 @@ in
       optionalString hasBonds "options bonding max_bonds=0";
 
     boot.kernel.sysctl = {
+      "net.ipv4.conf.all.forwarding" = mkDefault (any (i: i.proxyARP) interfaces);
       "net.ipv6.conf.all.disable_ipv6" = mkDefault (!cfg.enableIPv6);
       "net.ipv6.conf.default.disable_ipv6" = mkDefault (!cfg.enableIPv6);
       "net.ipv6.conf.all.forwarding" = mkDefault (any (i: i.proxyARP) interfaces);
     } // listToAttrs (flip concatMap (filter (i: i.proxyARP) interfaces)
-        (i: forEach [ "4" "6" ] (v: nameValuePair "net.ipv${v}.conf.${replaceChars ["."] ["/"] i.name}.proxy_arp" true)))
+        (i: [(nameValuePair "net.ipv4.conf.${replaceChars ["."] ["/"] i.name}.proxy_arp" true)
+             (nameValuePair "net.ipv6.conf.${replaceChars ["."] ["/"] i.name}.proxy_ndp" true)]))
```

Comment:
It seems GitHub ate my comment.

This reads like we always set a wrong proxy_ndp sysctl (so it never worked), and even if we set it now, the kernel-provided ndp proxy infrastructure requires addition of each individual address to ndp proxy for (see https://github.com/DanielAdolfsson/ndppd/blob/master/README#L57)

It seems more useful to just remove the NDP-specific bits from the `proxyARP` documentation, and let users either use ndppd (which is packaged in NixOS), or set options themselves.

Luckily, as it never worked before, we don't break any existing configurations.

Reply:
Hmm...you are right. Setting proxy_ndp alone is not enough to make this work. 

The question for me here would be whether a user who sets this knows that she has to additionally set something like `ip -6 neigh add proxy <ip> dev <if>` or not. If she is aware of this then this PR is aligned with her expectations, if not than obviously not. 

In any way, another way to go apart from your suggestion would be to generate a warning message that tells users that additional settings have to be specified for this to work and alternatively point to ndppd.

Reply:
`proxyARP` shouldn't enable proxy NDP - these are two different concepts, and we shouldn't mix them. Given it never worked, I don't see a problem with removing the (disfunctional and dangerous) proxy NDP part.

Reply:
Agreed. We shouldn't mix the two. It's better to force users to make this decision consciously.

Reply:
Can you push a new commit changing this?

Reply:
Changed this.